### DISCUSSION TITLE
features/publish-schema

### DISCUSSION THREAD

**Discussion by giscus** [graphql-hive/console#1282] [Docs Discussions]
# features/publish-schema

GraphQL Hive Documentation

https://docs.graphql-hive.com/features/publish-schema

**Comment by mausch** [comment]
```
hive schema:publish --service reviews --url http://my-service.com/graphql schema.graphql
```

What's the `url` parameter for? What does it do?

**Reply by dotansimha** [reply]
Hi @mausch ! When you are building a GraphQL API in a Federated architecture (using stitching/Apollo Federation), you need to provide a URL of where the service is running. The URL will later be exposed as part of the metadata/Supergraph, and this is how the gateway knows where to fetch data from.

**Reply by mausch** [reply]
Hi, I know what stitching is but that still doesn't quite explain what the `url` is for exactly. Is it just used for metadata purposes or will Hive want to make actual requests against it?

**Reply by dotansimha** [reply]
Hive will not make any requests against it, it's just metadata used to build your Federated graph.
